hikari/traits.py

```diff
@@ -60,42 +57,64 @@
     from hikari.api import rest as rest_
     from hikari.api import shard as gateway_shard
     from hikari.api import voice as voice_
-    from hikari.events import base_events  # noqa F401 - Unused (False positive)
 
-EventT_co = typing.TypeVar("EventT_co", bound="base_events.Event", covariant=True)
-"""Type-hint for a covariant event implementation instance.
+    _T = typing.TypeVar("_T")
 
-This will bind to the bound event type, or any subclass of that type.
-"""
+_Protocol = NotImplemented
 
-EventT_inv = typing.TypeVar("EventT_inv", bound="base_events.Event")
-"""Type-hint for an invariant event implementation instance.
 
-This will bind to the bound event type only. Subclasses and superclasses will
-not be matched.
-"""
+class _FastProtocol(abc.ABCMeta):
```

Comment:
Would it be worth it also defining subclass check behaviour here?

Reply:
I would rather keep that to the default

Reply:
Except wouldn't the default behaviour not match up with protocol subclass behaviour?

Reply:
I have implemented all it seems to do into the creation of the class. If I'm missing something, let me know

Reply:
Except this won't have the subclass behaviour
```python
>>> import typing
>>> @typing.runtime_checkable
class FooProto(typing.Protocol):
	def foo():
		...

>>> class Foo:
	def foo():
		...

>>> issubclass(Foo, FooProto)
True
>>> @typing.runtime_checkable
class FooProto(typing.Protocol):
	bar: int
	def foo():
		    ...

		    
>>> class Foo:
	def bar():
		...
	def foo():
		...

		
>>> issubclass(Foo, FooProto)
Traceback (most recent call last):
  File "<pyshell#27>", line 1, in <module>
    issubclass(Foo, FooProto)
  File "C:\Python38\lib\abc.py", line 102, in __subclasscheck__
    return _abc_subclasscheck(cls, subclass)
  File "C:\Python38\lib\typing.py", line 1081, in _proto_hook
    raise TypeError("Protocols with non-method members"
TypeError: Protocols with non-method members don't support issubclass()
```